### DISCUSSION TITLE
Explainability for STGCN Models

### DISCUSSION THREAD

**Discussion by sencer4898** [pyg-team/pytorch_geometric#8090] [Q&A]
We are working on spatiotemporal graph data (human skeleton extracted from videos), where graphs (videos) are classified into a class. The model we use is implemented with Conv2D layers of Pytorch. I wanted to get explanations for specific samples using GNNExplainer of pytorch_geom, hoping that it would work only by manipulating edge_index. However, I found out that it is compatible only with models (layers) from pytorch_geom. Also, it expects an NxF (spatial) feature tensor and gives dimension error for TxNxF (spatiotemporal). I have two questions:

1. Can an STGCN model written with Pytorch layers be explained using an explainer from pytorch_geom? (Any workarounds)
2. If not, can you suggest a different method/library giving explanations similar to GNNExplainer?

Thanks in advance
P.S. We need important edges of the adjacency matrix (spatial connections); importance calculation for temporal connections is not essential.

**Comment by rusty1s** [comment]
1. `GNNExplainer` can explain any models that utilizes `MessagePassing` layers. I am not sure if the `STGCN` model does that.
2. An alternative is to use `CaptumExplainer` which doesn't have this requirement. This should work for any model (hopefully).
